sdks/python/apache_beam/ml/anomaly/thresholds.py

```diff
@@ -17,172 +17,13 @@
-  and updating the prediction labels within `AnomalyResult` objects.
-
-  Args:
-    threshold_fn_spec (Spec): Specification defining the `ThresholdFn` to be
-      used.
-  """
-  def __init__(self, threshold_fn_spec: Spec):
-    self._threshold_fn_spec = threshold_fn_spec
-    self._threshold_fn = None
-
-  def _apply_threshold_to_predictions(
-      self, result: AnomalyResult) -> AnomalyResult:
-    """Updates the prediction labels in an AnomalyResult using the ThresholdFn.
-
-    Args:
-      result (AnomalyResult): The input `AnomalyResult` containing anomaly
-        scores.
-
-    Returns:
-      AnomalyResult: A new `AnomalyResult` with updated prediction labels
-        and threshold values.
-    """
-    predictions = [
-        dataclasses.replace(
-            p,
-            label=self._threshold_fn.apply(p.score),
-            threshold=self._threshold_fn.threshold) for p in result.predictions
-    ]
-    return dataclasses.replace(result, predictions=predictions)
-
-
-class StatelessThresholdDoFn(BaseThresholdDoFn):
-  """Applies a stateless ThresholdFn to anomaly detection results.
-
-  This DoFn is designed for stateless `ThresholdFn` implementations. It
-  initializes the `ThresholdFn` once during setup and applies it to each
-  incoming element without maintaining any state across elements.
-
-  Args:
-    threshold_fn_spec (Spec): Specification defining the `ThresholdFn` to be
-      used.
-
-  Raises:
-    AssertionError: If the provided `threshold_fn_spec` leads to the
-      creation of a stateful `ThresholdFn`.
-  """
-  def __init__(self, threshold_fn_spec: Spec):
-    threshold_fn_spec.config["_run_init"] = True
-    self._threshold_fn: Any = Specifiable.from_spec(threshold_fn_spec)
-    assert not self._threshold_fn.is_stateful, \
-      "This DoFn can only take stateless function as threshold_fn"
-
-  def process(self, element: Tuple[Any, Tuple[Any, AnomalyResult]],
-              **kwargs) -> Iterable[Tuple[Any, Tuple[Any, AnomalyResult]]]:
-    """Processes a batch of anomaly results using a stateless ThresholdFn.
-
-    Args:
-      element (Tuple[Any, Tuple[Any, AnomalyResult]]): A tuple representing
-        an element in the Beam pipeline. It is expected to be in the format
-        `(key1, (key2, AnomalyResult))`, where key1 is the original input key,
-        and key2 is a disambiguating key for distinct data points.
-      **kwargs:  Additional keyword arguments passed to the `process` method
-        in Beam DoFns.
-
-    Yields:
-      Iterable[Tuple[Any, Tuple[Any, AnomalyResult]]]: An iterable containing
-        a single output element with the same structure as the input, but with
-        the `AnomalyResult` having updated prediction labels based on the
-        stateless `ThresholdFn`.
-    """
-    k1, (k2, result) = element
-    yield k1, (k2, self._apply_threshold_to_predictions(result))
-
-
-class StatefulThresholdDoFn(BaseThresholdDoFn):
```

Comment:
Is there a reason to move this out of thresholds.py? It seems more natural here.

If it is just a refactor, could we split that into a separate PR? This one is already quite large